compiler/src/codegen/transl_anf.re

```diff
@@ -825,7 +835,23 @@ let rec compile_comp = (~id=?, env, c) => {
       MAllocate(MInt32(Int64.to_int32(n)))
     | CNumber(Const_number_int(n)) => MAllocate(MInt64(n))
     | CNumber(Const_number_float(f)) => MAllocate(MFloat64(f))
-    | CNumber(Const_number_rational(n, d)) => MAllocate(MRational(n, d))
+    | CNumber(Const_number_rational(neg, n, d, _, _)) =>
+      failwith("compile_comp: rational after ANF")
```

Comment:
So, because the rationals have bigints as their underlying values now, the operands to `Const_number_rational` are no longer immediates. During the ANF phase, we translate these constants to the new `CRational` type, which has the anf-ified (immediate) operands as its arguments. Once we are in the ANF->Mashtree step, we should not have any unprocessed `Const_number_rational`s left.

Reply:
That doesn't work here, since it violates ANF. A rational literal is _not_ a cexpr, since you have to perform the intermediate let-binds for the numerator and denominator. This translation would arguably be even more convoluted to add here than in linearize, since we would need to special-case up in `compile_anf_expr`, not `compile_comp`, thereby increasing the number of points of failure. I really think that implementing this in `linearize` is the cleanest solution. I can add a comment if that makes it more palatable.

Reply:
Just for the GitHub history: we chatted offline and decided that it made sense to attempt this refactor and go with it if it ended up not being too convoluted. I made those changes and pushed them to this branch.